python.qs

```diff
@@ -0,0 +1,191 @@
+            console.log(python_info);
+            if (python_info[1] == 0) {
+              widget.pythonInfo.html = "<p><strong style='color: green;'>OK</p><p>" + python_info[0].replace('\n', '<br />') + "</p>";
+              widget.complete = true;
+              installer.setValue("PythonPath", text);
+              return;
+            } else {
+              widget.pythonInfo.html = "<p><strong>Error calling Python:</strong></p><p>Return Code: " + python_info[1] + "</p><p>" + python_info[0] + "</p>";
+            }
+        } else {
+          widget.pythonInfo.html = "<p><strong>Python not found:</strong></p>";
+        }
+
+    }
+}
+
+Component.prototype.chooseTarget = function () {
+    var widget = gui.pageWidgetByObjectName("DynamicPythonToolsWidget");
+    if (widget != null) {
+
+       var fileName = QFileDialog.getOpenFileName("Python Executable",
+                                                widget.targetDirectory.text,
+                                                "Executable (*)");
+        if (fileName != "") {
+            widget.targetDirectory.text = Dir.toNativeSparator(fileName);
+        }
+    }
+}
+
+Component.prototype.pythonToolsWidgetEntered = function()
+{
+    var widget = gui.pageWidgetByObjectName("DynamicPythonToolsWidget");
+    var page = gui.pageByObjectName("DynamicPythonToolsWidget");
+
+    // var component = installer.componentByName("PythonTools");
+    if (!component.installationRequested()) {
+      console.log("No Python Tools Installation required")
+      if (page != null) {
+        installer.removeWizardPage(component, "PythonToolsWidget");
+      }
+      // page.hide();
+      // widget.complete = true;
+      return;
+    }
+
+    console.log("Python Tools Installation required")
+    if (page != null) {
+      page.show();
+      return;
+    }
+
+    if (installer.addWizardPage(component, "PythonToolsWidget", QInstaller.ReadyForInstallation)) {
+        var widget = gui.pageWidgetByObjectName("DynamicPythonToolsWidget");
+        if (widget != null) {
+            widget.targetChooser.clicked.connect(this, Component.prototype.chooseTarget);
+            widget.targetDirectory.textChanged.connect(this, Component.prototype.targetChanged);
+
+            widget.windowTitle = "Python Tools";
+
+            var sep = ":";
+            if (systemInfo.productType === "windows") {
+              sep = ";"
+            }
+            var paths = installer.environmentVariable("PATH").split(sep);
+
+            console.log(paths);
+            var pythonpath = installer.findPath("python", paths);
+            // Component.prototype.targetChanged(pythonpath);
+
+            widget.targetDirectory.text = Dir.toNativeSparator(pythonpath); // installer.value("TargetDir"));
+        }
+
+        var page = gui.pageByObjectName("DynamicPythonToolsWidget");
+        if (page != null) {
+        //  page.entered.connect(this, Component.prototype.pythonToolsWidgetEntered);
+        }
+    }
+}
```

Comment:
Toggle page based on selection of component. Could probably be a enhanced a bit.

Reply:
I noticed a little funny business when I would advance the installer page and then go back and unselect the python tools option.  Probably a corner case, as long as you just advance the installer forward it seemed to be fine.